lib/utilities/firebase_service.dart

```diff
@@ -40,17 +43,28 @@ class FirebaseService {
 
   // Register a user for an event
   Future<void> registerForEvent(String eventId, String userId) async {
-    try {
-      final eventDoc = _firestore.collection('events-collection').doc(eventId);
-      await eventDoc.update({
-        'registeredUsers': FieldValue.arrayUnion([userId]),
-        'slotsAvailable': FieldValue.increment(-1), // Decrement available slots
-      });
-      print('User $userId registered for event $eventId');
-    } catch (e) {
-      print('Error registering for event: $e');
-    }
+  try {
+    final eventDoc = _firestore.collection('events-collection').doc(eventId);
+
+    // Add user to the event's registered users list and decrement available slots
+    await eventDoc.update({
+      'registeredUsers': FieldValue.arrayUnion([userId]),
+      'slotsAvailable': FieldValue.increment(-1), // Decrement available slots
```

Comment:
This value has a similar issue than the previous one. What if this value says 4 available spaces but we have 8 out of 8 registrations in the registration collection? Then we would not know which one is right. I think it's fine for know but maybe a better approach would have been to populate this field by querying the number of registrations that match the eventId and subtracting them from the total number of slots.